File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        low = 0
        high = x
        
        while low <= high:
            mid = (high + low)//2
            if (mid ** 2) > x:
                high = mid-1
            
            elif (mid ** 2) < x:
                low = mid+1
            
            else:
                return mid
        
        return high
            
        

File: 0069-sqrtx/0069-sqrtx.go
import (
    "fmt"
    "math"
)

func mySqrt(x int) int {
    var left = -1;
    var right = 50000;
    
    for right > left + 1 {
        mid := left + ((right - left) / 2);

        if square := int(math.Pow(float64(mid), 2)); square <= x {
            left = mid;
        } else {
            right = mid;
        }
    }

    return left;
}
